src/vs/workbench/contrib/authentication/browser/authentication.contribution.ts

```diff
@@ -147,22 +147,36 @@ class AuthenticationExtensionsContribution extends Disposable implements IWorkbe
 	}
 
 	private _cleanupRemovedExtensions(removedExtensions?: readonly IExtensionDescription[]): void {
-		const extensionIdsToRemove = removedExtensions
-			? new Set(removedExtensions.map(e => e.identifier.value))
-			: new Set(this._extensionService.extensions.map(e => e.identifier.value));
-
 		// If we are cleaning up specific removed extensions, we only remove those.
 		const isTargetedCleanup = !!removedExtensions;
+		
+		let extensionIdsToRemove: Set<string>;
+		if (isTargetedCleanup) {
+			// For targeted cleanup, remove only the specified extensions
+			extensionIdsToRemove = new Set(removedExtensions!.map(e => e.identifier.value));
+		} else {
+			// For general cleanup, we need to find stored extensions that are no longer installed
+			const installedExtensionIds = new Set(this._extensionService.extensions.map(e => e.identifier.value));
+			extensionIdsToRemove = new Set<string>();
+			
+			// Find all stored extension IDs that are no longer installed
+			const providerIds = this._authenticationQueryService.getProviderIds();
```

Comment:
Besides readability, tell me how this change is different than before, if at all

Reply:
Isn't that equivalent?

I'll admit, in original, `extensionIdsToRemove` was a bad name... it should have been named something generic like `extensionIdSet`.

But original basically takes the set of all extensions in the extension service, and when we iterate over extensions in the auth service we remove the ones that are not in that set.

And the new implementation builds a set of extensions that are present in the auth service but not in the extension service... and then removes those.

Reply:
So, Matt reported that the Microsoft authentication extension's access state was being removed... even though it definitely should be available and installed because it's a built-in extension.

This leads me to believe that maybe this ExtenionService API does not include built-in extensions...? Can you check on this?